minio-mixin/minio_v1.libsonnet

```diff
@@ -0,0 +1,97 @@
+local g = import 'grafana-builder/grafana.libsonnet';
+
+local panel_settings_bytes = {
+  yaxes: g.yaxes('bytes'),
+};
+
+local panel_settings_qps = {
+  stack: true,
+};
+
+// Add common filters
+local f(s) = s % 'instance=~"$instance", job=~"$job"';
+
+g.dashboard('MinIO distributed cluster metrics', std.md5('minio_v1'))
+.addTemplate('job', 'minio_version_info', 'job')
+.addMultiTemplate('instance', 'minio_version_info{job="$job"}', 'instance')
+.addMultiTemplate('disk', 'disk_storage_available{job="$job"}', 'disk')
+.addRow(
+  g.row('Overview')
+  .addPanel(
+    g.panel('Storage Used') +
+    g.statPanel('sum(disk_storage_used{disk=~"$disk", job=~"$job"}) by (disk) \n                   / sum(disk_storage_total{disk=~"$disk", job=~"$job"}) by (disk)') +
+    {
+      type: 'gauge',
+      // The default 'percentunit' format of statPanel() doesn't seem to have an effect
+      fieldConfig: {
+        defaults: {
+          unit: 'percentunit',
+        },
+      },
+    }
+  )
+  .addPanel(
+    g.panel('Disks Total') +
+    g.statPanel(f('sum(minio_disks_total{%s})'), 'none')
+  )
+  .addPanel(
+    g.panel('Disks Offline') +
+    g.statPanel(f('sum(minio_disks_offline{%s})'), 'none')
+  )
+  .addPanel(
+    g.panel('Errors') +
+    g.queryPanel(f('s3_errors_total{%s}'), '')
+  )
+)
+.addRow(
+  g.row('Storage')
+  .addPanel(
+    g.panel('Storage Used') +
+    g.queryPanel(f('disk_storage_used{disk=~"$disk",%s}'), '') +
+    panel_settings_bytes,
+  )
+  .addPanel(
+    g.panel('Storage Available') +
+    g.queryPanel(f('disk_storage_available{disk=~"$disk",%s}'), '') +
+    panel_settings_bytes,
+  )
+  .addPanel(
+    g.panel('Storage Total') +
+    g.queryPanel(f('disk_storage_total{disk=~"$disk",%s}'), '') +
+    panel_settings_bytes,
+  )
+)
+.addRow(
+  g.row('Requests')
+  .addPanel(
+    g.panel('Read rps') +
+    g.queryPanel(f('sum(rate(s3_requests_total{api=~"get.*|list.*|head.*",%s}[$__rate_interval])) by (api)'), '{{api}}') +
+    panel_settings_qps
+  )
+  .addPanel(
+    g.panel('Write rps') +
+    g.queryPanel(f('sum(rate(s3_requests_total{api=~"put.*",%s}[$__rate_interval])) by (api)'), '{{api}}') +
+    panel_settings_qps
+  )
+  .addPanel(
+    g.panel('Delete rps') +
+    g.queryPanel(f('sum(rate(s3_requests_total{api=~"delete.*",%s}[$__rate_interval])) by (api)'), '{{api}}')
+  )
+)
+.addRow(
+  g.row('Performance')
+  .addPanel(
+    g.panel('Read latency') +
+    g.latencyPanel('s3_ttfb_seconds', f('{api=~"get.*|list.*|head.*",%s}'))
+  )
+  .addPanel(
+    g.panel('Internode traffic') +
```

Comment:
Let's at least include a note on this indicating that it's expected to be zero if running in a single-node configuration.

Reply:
Sure thing.  Are you thinking a comment in the code here, or something in the dashboard like a panel description?

Reply:
Sorry, ambiguous - yeah, i was thinking something terse in the panel description.